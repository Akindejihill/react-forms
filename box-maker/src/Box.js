const Box = (props) => {
    const removeBox = () => {
        props.removeBox(props.id);
    };

    return (
        <div style={{backgroundColor: props.backgroundColor, width: `${props.width}px`, height: `${props.height}px` }} >
            <button onClick={removeBox}>X</button>
        </div>
    );
};

export default Box;
