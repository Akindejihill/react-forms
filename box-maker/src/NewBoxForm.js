import React, { useState } from "react";


const NewBoxForm = ({ addBox }) => {
  const [width, setWidth] = useState("100");
  const [height, setHeight] = useState("100");
  const [backgroundColor, setBackgroundColor] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();
    addBox({ id: Date.now(), width: Number(width), height: Number(height), backgroundColor });
    setWidth("");
    setHeight("");
    setBackgroundColor("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="width">Width:</label>
      <input type="number" id="width" value={width} onChange={(e) => setWidth(e.target.value)} />

      <label htmlFor="height">Height:</label>
      <input type="number" id="height" value={height} onChange={(e) => setHeight(e.target.value)} />

      <label htmlFor="backgroundColor">Background Color:</label>
      <input type="text" id="backgroundColor" value={backgroundColor} onChange={(e) => setBackgroundColor(e.target.value)} />

      <button>Add Box</button>
    </form>
  );
};

export default NewBoxForm;