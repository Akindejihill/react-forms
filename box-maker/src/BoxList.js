import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
    const [boxes, setBoxes] = useState([]);

    /** used by the NewBoxForm component
     * adds a box description to the list 
     * of box descriptions in the 'boxes' state
    */
    const addBox = (box) => {
        setBoxes([...boxes, box]);
    };

    const removeBox = (id) => {
        setBoxes(boxes.filter((box) => box.id !== id));
    };

    /**
     * creates box components based on parameters in 'boxes' state
     */ 
    const renderBoxes = () => {
        return boxes.map((box) => (
            <Box
                key={box.id}
                id={box.id}
                width={box.width}
                height={box.height}
                backgroundColor={box.backgroundColor}
                removeBox={removeBox}
            />
        ));
    };

    return (
        <div>
            <NewBoxForm addBox={addBox} />
            {renderBoxes()}
        </div>
    );
};

export default BoxList;
