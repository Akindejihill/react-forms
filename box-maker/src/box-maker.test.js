import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";
import BoxList from "./BoxList";
import NewBoxForm from "./NewBoxForm";

//Smoke test
it("renders without crashing", () => {
    const { getByText } = render(
        <Box backgroundColor="red" width="100" height="100" />
    );
    const boxElement = getByText("X");

    expect(boxElement).toBeInTheDocument();
});

it("renders without crashing", () => {
    render(<BoxList />);
});

it("renders without crashing", () => {
    render(<NewBoxForm />);
});

//Snapshots
it("matches snapshot", function () {
    const { asFragment } = render(
        <Box backgroundColor="red" width="100" height="100" />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});


