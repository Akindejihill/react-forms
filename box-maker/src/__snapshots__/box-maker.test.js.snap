// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: red; width: 100px; height: 100px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        type="number"
        value="100"
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        type="number"
        value="100"
      />
      <label
        for="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 3`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      type="number"
      value="100"
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      type="number"
      value="100"
    />
    <label
      for="backgroundColor"
    >
      Background Color:
    </label>
    <input
      id="backgroundColor"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
